include/Scale.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Scale.h - holds information about a scale and its intervals
+ *
+ * Copyright (c) 2020 Martin Pavelek <he29.HS/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef SCALE_H
+#define SCALE_H
+
+#include <cmath>
+#include <cstdint>
+#include <vector>
+#include <QObject>
+#include <QString>
+
+#include "SerializingObject.h"
+
+class Interval : public SerializingObject
+{
+public:
+	Interval() : m_numerator(1), m_denominator(1), m_cents(0), m_ratio(1) {};
+	explicit Interval(float cents);
+	Interval(uint32_t numerator, uint32_t denominator);
+
+	float getRatio() const {return m_ratio;}
+
+	QString getString() const
+	{
+		if (m_denominator) {return QString::number(m_numerator) + "/" + QString::number(m_denominator);}
+		else {return QString().sprintf("%.4f", m_cents);}
```

Comment:
[`QString::sprintf`](https://doc.qt.io/qt-5/qstring-obsolete.html#sprintf) is deprecated.